docs/src/appendix/cheatcodes/mock_call_when.md

```diff
@@ -0,0 +1,43 @@
+# `mock_call_when`
+
+Cheatcodes mocking contract entry point calls:
```

Comment:
It's the same as description of `mock_call`. Maybe let's emphasize the difference?

Reply:
Updated with `Cheatcodes mocking contract entry point calls based on calldata:`